integration/test/Test/Teams.hs

```diff
@@ -276,3 +280,51 @@ testUpgradePersonalToTeamAlreadyInATeam = do
   bindResponse (upgradePersonalToTeam alice "wonderland") $ \resp -> do
     resp.status `shouldMatchInt` 403
     resp.json %. "label" `shouldMatch` "user-already-in-a-team"
+
+testTeamMemberCsvExport :: (HasCallStack) => App ()
+testTeamMemberCsvExport = do
+  (owner, tid, members) <- createTeam OwnDomain 10
+  let numClients = [0, 1, 2] <> repeat 0
+  modifiedMembers <- for (zip numClients (owner : members)) $ \(n, m) -> do
+    handle <- randomHandle
+    putHandle m handle >>= assertSuccess
+    replicateM_ n $ addClient m def
+    void $ I.putSSOId m def {I.scimExternalId = Just "foo"} >>= getBody 200
+    setField "handle" handle m
+      >>= setField "role" (if m == owner then "owner" else "member")
+      >>= setField "num_clients" (show n)
+
+  memberMap :: Map.Map String Value <- fmap Map.fromList $ for (modifiedMembers) $ \m -> do
+    uid <- m %. "id" & asString
+    pure (uid, m)
+
+  bindResponse (getTeamMembersCsv owner tid) $ \resp -> do
+    resp.status `shouldMatchInt` 200
+    let rows = sort $ tail $ B8.lines $ resp.body
+    length rows `shouldMatchInt` 10
+    for_ rows $ \row -> do
+      let cols = B8.split ',' row
+      let uid = read $ B8.unpack $ cols !! 11
+      let mem = memberMap Map.! uid
+
+      ownerId <- owner %. "id" & asString
+      let ownerMember = memberMap Map.! ownerId
+
+      let parseField = unquote . read . B8.unpack . (cols !!)
+
+      parseField 0 `shouldMatch` (mem %. "name")
+      parseField 1 `shouldMatch` (mem %. "handle")
+      parseField 2 `shouldMatch` (mem %. "email")
+      role <- mem %. "role" & asString
+      parseField 3 `shouldMatch` role
+      when (role /= "owner") $ do
+        now <- formatTime defaultTimeLocale "%Y-%m-%d" <$> liftIO getCurrentTime
+        take 10 (parseField 4) `shouldMatch` now
+        parseField 5 `shouldMatch` (ownerMember %. "handle")
+      parseField 7 `shouldMatch` "wire"
+      parseField 9 `shouldMatch` "foo"
+      parseField 12 `shouldMatch` (mem %. "num_clients")
+  where
+    unquote :: String -> String
+    unquote ('\'' : x) = x
+    unquote x = x
```

Comment:
there could be more `'` in some of those strings.

```suggestion
    unquote :: String -> String
    unquote = init . tail
```

Reply:
I'm not sure what you're saying here. For whatever reason, we escape these values by adding a quote at the beginning when they start with some "forbidden character". It doesn't make a lot of sense to me, but that's what the exporter was doing before, so I didn't change it. This function simply undoes that escaping.

Reply:
sorry, i was reading it wrong. all good!